src/quo2/components/code/snippet.cljs

```diff
@@ -1,175 +1,110 @@
                  (render-nodes children))
            ;; Remove newlines as we already render each line separately.
-           (-> value string/trim-newline)))
+           (string/trim-newline value)))
        nodes))
 
+(defn- line
+  [{:keys [line-number line-number-width]} children]
+  [rn/view {:style style/line}
+   [rn/view {:style (style/line-number line-number-width)}
+    [text/text
+     {:style  (style/text-style ["line-number"])
+      :weight :code
+      :size   :paragraph-2}
+     line-number]]
+   children])
+
 (defn- code-block
   [{:keys [rows line-number-width]}]
-  [into [:<>]
+  [rn/view
    (->> rows
         (render-nodes)
-        ;; Line numbers
-        (map-indexed (fn [idx row]
-                       (conj [rn/view {:style {:flex-direction :row}}
-                              [rn/view
-                               {:style {:width        line-number-width
-                                        ;; 8+12 margin
-                                        :margin-right 20}}
-                               [text/text
-                                {:weight :code
-                                 :size   :paragraph-2
-                                 :style  (text-style ["line-number"])}
-                                (inc idx)]]]
-                             row))))])
+        (map-indexed (fn [idx row-content]
+                       [line
+                        {:line-number       (inc idx)
+                         :line-number-width line-number-width}
+                        row-content]))
+        (into [:<>]))])
 
-(defn- native-renderer
-  []
-  (let [text-height (reagent/atom nil)]
-    (fn [{:keys [rows max-lines on-copy-press]}]
-      (let [background-color      (colors/theme-colors
-                                   colors/white
-                                   colors/neutral-80-opa-40)
-            background-color-left (colors/theme-colors
-                                   colors/neutral-5
-                                   colors/neutral-80)
-            border-color          (colors/theme-colors
-                                   colors/neutral-20
-                                   colors/neutral-80)
-            rows                  (bean/->clj rows)
-            font-scale            (:font-scale (rn/use-window-dimensions))
-            max-rows              (or max-lines (count rows)) ;; Cut down on rows to process.
-            max-line-digits       (-> rows count (min max-rows) str count)
-            ;; ~ 9 is char width, 18 is width used in Figma.
-            line-number-width     (* font-scale (max 18 (* 9 max-line-digits)))
-            max-text-height       (some-> max-lines
-                                          (* font-scale 18)) ;; 18 is font's line height.
-            truncated?            (and max-text-height (< max-text-height @text-height))
-            maybe-mask-wrapper    (if truncated?
-                                    [masked-view/masked-view
```

Comment:
It was using a masked view over a linear-gradient, it was a good idea, the puprose of this was to add this gradient at the bottom of the component when the fragment is truncated:
![image](https://user-images.githubusercontent.com/90291778/230192763-90b9807f-eac3-4247-b78e-a007dd993ee7.png)

The problem was that mask was being applied to the full code view, and according to designs, that gradient is only 47 units height. The gradient was being applied from 75% of the view  to 100%, it looked ok on small fragments, but on larger ones it started to look different than designs.

I removed the use of a mask and just added a linear gradient drawn over the bottom of the fragment